system/security/plugins/htpasswdSecurity/htpasswdSecurity.cpp

```diff
@@ -25,12 +25,29 @@
 class CHtpasswdSecurityManager : public CBaseSecurityManager
 {
 public:
-    CHtpasswdSecurityManager(const char *serviceName, IPropertyTree *secMgrCfg, IPropertyTree *authConfig) : CBaseSecurityManager(serviceName, (IPropertyTree *)NULL)
+    CHtpasswdSecurityManager(const char *serviceName, IPropertyTree *secMgrCfg, IPropertyTree *bindConfig) : CBaseSecurityManager(serviceName, (IPropertyTree *)NULL)
 	{
         if (secMgrCfg)
             pwFile.set(secMgrCfg->queryProp("@htpasswdFile"));
         if(pwFile.isEmpty())
             throw MakeStringException(-1, "htpasswdFile not found in configuration");
+
+        Owned<IPropertyTree> authcfg = bindConfig->getPropTree("Authenticate");
```

Comment:
What if you wrap lines 35 - 41 in braces (in its own scope) so the property tree gets released before loading the next one. Same thing with the following one